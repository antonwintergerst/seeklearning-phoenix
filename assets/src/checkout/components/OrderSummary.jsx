import React from 'react';

const OrderSummary = ({ products }) => (
  <div className="ui basic segment ad-selection-form">
    <table className="ui celled striped table">
      <thead className="center aligned">
        <tr>
          <th>Product</th>
          <th>Qty</th>
          <th>Price</th>
        </tr>
      </thead>
      <tbody>
        {products.map(product => (
          <tr key={product.id}>
            <td>
              <div className="ui ribbon label">{product.name}</div>
            </td>
            <td className="center aligned">
              {product.quantity}
            </td>
            <td className="center aligned">
              {product.price}
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  </div>
);

export default OrderSummary;
