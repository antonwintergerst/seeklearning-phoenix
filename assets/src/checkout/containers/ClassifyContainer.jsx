import React from 'react';
// redux
import { connect } from 'react-redux';
import { setAdType, addProduct } from '../actions';

import AdSelectionForm from '../components/AdSelectionForm';
import { adTypes, adFeatures } from './ClassifyContainer.data';

const ClassifyContainer = (props) => {
  const { quote, job } = props;
  const adTypesState = adTypes.map(adType => ({ ...adType, isActive: job && job.adType.id === adType.id }));
  const onContinueClicked = () => {
    if (!job) {
      return;
    }
    props.addProduct(quote, job);
  };
  return (
    <div className="ui container">
      <h2 className="ui header">Choose your style of ad</h2>
      <AdSelectionForm adTypes={adTypesState} adFeatures={adFeatures} onSelectClicked={props.setAdType} />
      <button className="ui huge button pink continue" onClick={onContinueClicked}>Continue</button>
    </div>
  );
};

// Redux state mapping
const mapStateToProps = state => ({
  quote: state.checkout.quote,
  job: state.checkout.job,
});

export default connect(mapStateToProps, {
  setAdType,
  addProduct,
})(ClassifyContainer);
