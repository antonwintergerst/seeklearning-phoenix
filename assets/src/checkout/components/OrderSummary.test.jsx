import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import OrderSummary from './OrderSummary';
import { quote } from '../containers/ClassifyContainer.data';

describe('OrderSummary', () => {
  it('renders as expected', () => {
    const tree = ReactTestRenderer.create(<OrderSummary products={quote.products} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
