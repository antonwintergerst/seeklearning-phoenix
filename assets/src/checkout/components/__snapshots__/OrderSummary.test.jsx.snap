// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSummary renders as expected 1`] = `
<div
  className="ui basic segment ad-selection-form"
>
  <table
    className="ui celled striped table"
  >
    <thead
      className="center aligned"
    >
      <tr>
        <th>
          Product
        </th>
        <th>
          Qty
        </th>
        <th>
          Price
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <div
            className="ui ribbon label"
          >
            Classic
          </div>
        </td>
        <td
          className="center aligned"
        />
        <td
          className="center aligned"
        >
          275
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="ui ribbon label"
          >
            StandOut
          </div>
        </td>
        <td
          className="center aligned"
        />
        <td
          className="center aligned"
        >
          330
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="ui ribbon label"
          >
            Guaranteed Hire
          </div>
        </td>
        <td
          className="center aligned"
        />
        <td
          className="center aligned"
        >
          330
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
